cognite/neat/rules/issues/spreadsheet.py

```diff
@@ -0,0 +1,381 @@
+        )
+        if self.url:
+            output += f" For further information visit {self.url}"
+        return output
+
+
+@dataclass(frozen=True)
+class InvalidPropertyError(InvalidRowError):
+    sheet_name = "Properties"
+
+
+@dataclass(frozen=True)
+class InvalidClassError(InvalidRowError):
+    sheet_name = "Classes"
+
+
+@dataclass(frozen=True)
+class InvalidContainerError(InvalidRowError):
+    sheet_name = "Containers"
+
+
+@dataclass(frozen=True)
+class InvalidViewError(InvalidRowError):
+    sheet_name = "Views"
+
+
+@dataclass(frozen=True)
+class InvalidRowUnknownSheetError(InvalidRowError):
+    sheet_name = "Unknown"
+
+    actual_sheet_name: str
+
+    @classmethod
+    def from_pydantic_error(
+        cls, error: ErrorDetails, read_info_by_sheet: dict[str, SpreadsheetRead] | None = None
+    ) -> Self:
+        sheet_name, _, row, column, *__ = error["loc"]
+        reader = (read_info_by_sheet or {}).get(str(sheet_name), SpreadsheetRead())
+        return cls(
+            column=str(column),
+            row=reader.adjusted_row_number(int(row)),
+            actual_sheet_name=str(sheet_name),
+            type=error["type"],
+            msg=error["msg"],
+            input=error.get("input"),
+            url=str(url) if (url := error.get("url")) else None,
+        )
+
+    def dump(self) -> dict[str, Any]:
+        output = super().dump()
+        output["actual_sheet_name"] = self.actual_sheet_name
+        return output
+
+
+_INVALID_ROW_ERROR_BY_SHEET_NAME = {
+    cls_.sheet_name: cls_ for cls_ in InvalidRowError.__subclasses__() if cls_ is not InvalidRowError
+}
+
+
+@dataclass(frozen=True)
+class NonExistingContainerError(InvalidPropertyError):
+    description = "The container referenced by the property is missing in the container sheet"
+    fix = "Add the container to the container sheet"
+
+    container_id: ContainerId
+
+    def message(self) -> str:
+        return (
+            f"In {self.sheet_name}, row={self.row}, column={self.column}: The container with "
+            f"id {self.container_id} is missing in the container sheet."
+        )
+
+    def dump(self) -> dict[str, Any]:
+        output = super().dump()
+        output["container_id"] = self.container_id
+        return output
+
+
+@dataclass(frozen=True)
+class NonExistingViewError(InvalidPropertyError):
+    description = "The view referenced by the property is missing in the view sheet"
+    fix = "Add the view to the view sheet"
+
+    view_id: ViewId
+
+    def message(self) -> str:
+        return (
+            f"In {self.sheet_name}, row={self.row}, column={self.column}: The view with "
+            f"id {self.view_id} is missing in the view sheet."
+        )
+
+    def dump(self) -> dict[str, Any]:
+        output = super().dump()
+        output["view_id"] = self.view_id
+        return output
+
+
+@dataclass(frozen=True, order=True)
+class ClassNoPropertiesNoParentsWarning(ValidationWarning):
```

Comment:
There are couple of issues which depending on the use case can be warning or error. Is this possible with the current setup?

Reply:
Then, you would have to duplicate have one error and one warning version of the issue.